test/testutils.js

```diff
@@ -96,12 +110,24 @@ testUtils.fixtureSetup = function(content) {
       fixture.appendChild(node);
     });
   }
-  axe._tree = axe.utils.getFlattenedTree(fixture);
-  axe._selectorData = axe.utils.getSelectorData(axe._tree);
 
   return fixture;
 };
 
+/**
+ * Method for injecting content into a fixture and caching
+ * the flattened DOM tree (light and Shadow DOM together)
+ *
+ * @param {String|Node} content Stuff to go into the fixture (html or DOM node)
+ * @return HTMLElement
+ */
+testUtils.fixtureSetup = function(content) {
+  'use strict';
+  testUtils.injectIntoFixture(content);
+  axe.teardown();
```

Comment:
Test code that uses loops (like for `input` types) would do so by calling `querySelector` or `fixtureSetup` inside the loop without tearing down. Since the tree changed I figured it was acceptable to call teardown before setting up the tree.